lib/src/connection.dart

```diff
@@ -409,11 +416,11 @@ abstract class _PostgreSQLExecutionContextMixin
   int get queueSize => _queue.length;
 
   @override
-  Future<PostgreSQLResult> query(
+  Future<PostgreSQLResult?> query(
     String fmtString, {
-    Map<String, dynamic> substitutionValues,
-    bool allowReuse,
-    int timeoutInSeconds,
+    Map<String, dynamic>? substitutionValues,
+    bool allowReuse = true,
```

Comment:
Please remove all of these default values. It makes it much easier to develop a tool like `package:postgres_pool`